unitOneAssessmentPractice.js

```diff
@@ -3,39 +3,98 @@ let assert = require('assert')
 // Question One:
 
 // Write a function called double that doubles a number
-
-// Uncomment out the next line to test your solution
+function double(num) {
+  return num*2;
+}
+double(14);
+// // Uncomment out the next line to test your solution
 // runQ1Tests()
 
 // Question Two:
 
 // Write a function called containsSeven that returns whether or not a number has an sevens in it.
+function containsSeven(num) {
+ let newNum = num.toString();
+ for (let i = 0; i < newNum.length; i++) {
+   if (newNum[i] === "7") {
+    return true;
+   } else {
+    i++;
+   }
+ }
+}
+ containsSeven(5378);
 
 // Uncomment out the next line to test your solution
 // runQ2Tests()
 
 // Question Three:
 
 // Write a function called capitalizeTheAs the capitalizes all of the lowercase "a"s in a string leaving all other characters the same
+function capitalizeTheAs(str) {
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === "a") {
+    return str.replace("a", "A");
+    }
+  }
+}
+capitalizeTheAs("giraffe");
 
 // Uncomment out the next line to test your solution
 // runQ3Tests()
 
 // Question Four:
 // Write a function called largest that returns the largest value in an array
+function largest(arr) {
+  let newLargest = "";
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] > newLargest) {
+      newLargest = arr[i];
+    } else {
+      i++;
+    }
+  }
+  return newLargest;
+}
+
+largest([8,1,12,14,6]);
 
 // Uncomment out the next line to test your solution
 // runQ4Tests()
 
 // Question Five:
 // Write a function called average that returns the average of an array
+function average(arr) {
+  let sum = 0;
+  for (let i=0; i < arr.length; i++) {
+    sum += arr[i];
+  }
+  return (sum/arr.length);
+}
 
+average([2,4,6,8]);
 // Uncomment out the next line to test your solution
 // runQ5Tests()
 
 // Question Six:
 // Write a function called mode that returns the most frequently occurring number in an array
 // HINT: Use an object where the keys are the numbers, and the values are how many times they appear in the array.
+function mode(arr) {
```

Comment:
Here is the code that I wrote for your reference in case you want to take a look: 
It basically creates an object with each number as a key and saves their frequency as value corresponding to each key. Then next for loop checks for biggest value for each, and logs which ever key has the most repeated frequency. 

function mode(arr) {
  let obj = {}
  let mostFrequentValue = 0
  let mostFrequentNum

  for(let num of arr) {
    if(obj[num]) {
      obj[num]++
    } else {
      obj[num] = 1
    }
  }
  console.log(obj)
  for(let num in obj) {
    if(obj[num] > mostFrequentValue) {
      mostFrequentValue = obj[num]
      mostFrequentNum = num
    }
  }
  return mostFrequentNum
}

mode([2,4,6,8,4,10,10, 4, 2, 4, 2, 2, 2, 2]);

Reply:
i worked on the logic behind this with alejo and then understanding the actual code with amin. i am pushing my code for this now that i am more comfortable breaking down the problem